zerver/actions/user_groups.py

```diff
@@ -323,9 +343,37 @@ def bulk_add_members_to_user_groups(
         for user_group in user_groups
     )
 
+    subscriber_ids_for_streams = get_user_ids_for_streams({stream.id for stream in streams})
+    new_stream_metadata_user_ids = bulk_can_access_stream_metadata_user_ids(streams)
+    recent_traffic = get_streams_traffic({stream.id for stream in streams}, realm)
```

Comment:
It should be fairly common that this function is called in situations where the `streams` lists are empty. So we should check these functions for being noops in that case. For example, `get_streams_traffic` probably deserves a tweak to check if the input list is empty.

Reply:
It might be that Django does this for us automatically, though.